src/controllers/admin.disable.account.controller.ts

```diff
@@ -0,0 +1,59 @@
+import { Request, Response } from 'express';
+import AuthService from '../service/auth.service';
+import { accountDeactivationEmailTemplate } from '../helpers/EmailTemplates/accountDeactivationEmailTemplate';
+import { sendEmail } from '../helpers/sendEmail';
+import { config } from 'dotenv';
+import { resolve } from 'path';
+config({ path: resolve(__dirname, '../.env') });
+export const disableUserAccount = async (
+  req: Request,
+  res: Response,
+) => {
+  try {
+    const { user_id } = req.params;
+    const user = await AuthService.querySingleUser({ where: { user_id } });
+    if (!user) {
+      return res.status(404).json({ message: "User doesn't exist" });
+    }
+    if (user.verified == false) {
+      return res.json({ message: 'User is already not verified' });
+    }
+    user.verified = false;
+    await user.save();
+    interface UsersData {
+      username: string;
+    }
+    const defaultUserData: UsersData = {
+      username: user.username,
+    };
+    interface EmailOptions {
+      to: string | string[];
+      from: string;
+      subject: string;
+      template: (data: UsersData) => string;
+      attachmentPath?: string;
+      isVerificationEmail?: boolean;
+    }
+    const adminEmail = process.env.ADMIN_EMAIL;
+    if (!adminEmail) {
+      throw new Error('Admin email is not defined');
+    }
+    const emailOptions: EmailOptions = {
+      to: user.email,
+      from: adminEmail,
+      subject: 'Deactivation',
+      template: (data: UsersData) =>
+        accountDeactivationEmailTemplate(
+          data.username || defaultUserData.username,
+        ),
+    };
+    sendEmail(emailOptions);
+    return res.status(200).json({
+      message: `User account disabled successfully`,
+    });
+  } catch (error) {
+    console.error('Error disabling user account:', error);
+    return res.status(500).json({ message: 'Internal server error' });
+  }
+};
```

Comment:
I haven't seen you disable the account anywhere in this file nor have i seen you use the appropriate service to do it!!!